queries/smali/highlights.scm

```diff
@@ -4,52 +4,106 @@
 
 (primitive_type) @type.builtin
 
-(array_type
-  "[" @punctuation.bracket)
+((class_identifier) @type.builtin
+  (#lua-match? @type.builtin "^Landroid/"))
+((class_identifier) @type.builtin
+  (#lua-match? @type.builtin "^Lcom/android/"))
+((class_identifier) @type.builtin
+  (#lua-match? @type.builtin "^Ldalvik/"))
+((class_identifier) @type.builtin
+  (#lua-match? @type.builtin "^Ljava/"))
 
 ; Methods
 
-(method_declaration
-  (method_identifier) @method)
+(method_definition
+  (method_signature (method_identifier) @method))
 
-(statement
+(expression
   (opcode) @_invoke
-  (full_method_identifier
-    (method_identifier) @method.call)
+  (value
+	(body
+	  (full_method_signature
+        (method_signature (method_identifier) @method.call))))
   (#lua-match? @_invoke "^invoke"))
 
-(method_identifier
-  "<init>(" @constructor
-  ")" @constructor)
+(method_handle
+  (full_method_signature
+	(method_signature (method_identifier) @method.call)))
+
+(call_site) @method.call
+
+(custom_invoke
+  (method_signature (method_identifier) @method.call))
+
+(annotation_value
+  (body
+    (method_signature (method_identifier) @method.call)))
+
+(annotation_value
+  (body
+    (full_method_signature
+      (method_signature (method_identifier) @method.call))))
+
+(field_definition
+  (value
+	(body
+		(method_signature (method_identifier) @method.call))))
 
-(method_identifier
-  "<clinit>(" @constructor
-  ")" @constructor)
+(field_definition
+  (value
+	(body
+	  (full_method_signature
+		(method_signature (method_identifier) @method.call)))))
+
+((method_signature
+  (method_identifier) @constructor)
+  (#any-of? @constructor "<init>" "<clinit>"))
 
 ; Fields
 
 (field_identifier) @field
 
-; Parameters
-
-(parameter) @parameter
+(annotation_key) @field
 
 ; Variables
 
-(variable) @variable
+(variable) @variable.builtin
+
+(local_directive
+  (identifier) @variable)
+
+; Parameters
+
+(parameter) @parameter ; more like @parameter.builtin but that doesn't exist in nvim-treesitter
+(param_identifier) @parameter
 
 ; Labels
 
-(label) @label
+[
+  (label)
+  (jmp_label)
+] @label
 
 ; Operators
 
 (opcode) @keyword.operator
 
 ((opcode) @keyword.return
- (#lua-match? @keyword.return "^return"))
+  (#lua-match? @keyword.return "^return"))
+
+((opcode) @conditional
+  (#lua-match? @conditional "^if"))
+((opcode) @conditional
+  (#lua-match? @conditional "^cmp"))
 
-"=" @operator
+((opcode) @exception
+  (#lua-match? @exception "^throw"))
+
+[
+  "="
+  ".."
+  "@"
```

Comment:
Why is `@` an operator?

Reply:
```smali
    invoke-custom {v0, v1}, call_site_0("doSomething", (LCustom;Ljava/lang/String;)Ljava/lang/String;, "just testing")@LBootstrapLinker;->backwardsLink(Ljava/lang/invoke/MethodHandles$Lookup;Ljava/lang/String;Ljava/lang/invoke/MethodType;Ljava/lang/String;)Ljava/lang/invoke/CallSite;

    invoke-custom {v0, v1}, call_site_2("doSomething", (LCustom;Ljava/lang/String;)Ljava/lang/String;, invoke-static@Lnonsense;->somemethod()V)@LBootstrapLinker;->normalLink(Ljava/lang/invoke/MethodHandles$Lookup;Ljava/lang/String;Ljava/lang/invoke/MethodType;Ljava/lang/invoke/MethodHandle;)Ljava/lang/invoke/CallSite;
```

very niche statements in Smali

Reply:
Looks more like `@punctuation.delimiter` to me.

Reply:
mmm yeah, its to separate targets and their types so that makes more sense to me too then